Test_Engine/Compute/ToJsonWithChecks.cs

```diff
@@ -0,0 +1,143 @@
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.Engine.Diffing;
+using BH.oM.Base;
+using BH.oM.Base.Attributes;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+
+namespace BH.Engine.Test
+{
+    public static partial class Compute
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Runs through the provided objects and tries to convert to Json. Then performs checks if the object can be deserialised, and if the de-serailised object is equal to the serialised version.")]
+        [Input("objects", "Objects to try to convert ToJson.")]
+        [MultiOutput(0, "validJson", "Json strings that pass all checks.")]
+        [MultiOutput(1, "failingToJson", "Json strings show some failure of running the ToJson convert.")]
+        [MultiOutput(2, "failingToJsonObject", "Obejcts that failed the ToJson convert.")]
+        [MultiOutput(3, "failingFromJson", "Json for cases where the FromJson is failing.")]
+        [MultiOutput(4, "failingFromJsonObjects", "Objects that had some failures going FromJson.")]
+        [MultiOutput(5, "notEqualJson", "Jsons strings for objects not equal to incoming object after serialisation ToJson followed by deserialisation FromJson.")]
+        [MultiOutput(6, "notEqualObjects", "Objects not equal to incoming object after serialisation ToJson followed by deserialisation FromJson.")]
+        public static Output<List<string>, List<string>, List<object>, List<string>, List<object>, List<string>, List<object>> TryToJsonAndCheckIfEqual(List<object> objects)
```

Comment:
Method name is a bit confusing. Am I correct in describing that what it does is serialise => deserialise => compare? If so, maybe a combination of the 3 words would be more representative?

Reply:
WIll rename